llvm/lib/Transforms/Scalar/EarlyCSE.cpp

```diff
@@ -1561,6 +1646,31 @@ bool EarlyCSE::processNode(DomTreeNode *Node) {
       continue;
     }
 
+    // Compare GEP instructions based on offset.
+    if (GEPValue::canHandle(&Inst)) {
+      auto *GEP = cast<GetElementPtrInst>(&Inst);
+      APInt Offset = APInt(SQ.DL.getIndexTypeSizeInBits(GEP->getType()), 0);
+      GEPValue GEPVal(GEP, GEP->accumulateConstantOffset(SQ.DL, Offset)
+                               ? Offset.tryZExtValue()
```

Comment:
I think the offset should be `int64_t` and use `trySExtValue()` here. GEP offsets are signed. (Mainly for clarify, I think it will not make a functional difference?)

Reply:
Yes, it's more clear to use `int64_t`.